modules/IsometricLayout/src/main/java/org/alexandrebarao/isometriclayout/IsometricLayout.java

```diff
@@ -0,0 +1,442 @@
+        double maxZvalueFromColumnZ=0; 
+        
+        for (int i = 0 ; i< nodeCount; i++ ) {
+                   Node n = nodes[i]; 
+                   
+                   if ( columnZ != null ) {
+                        double zV ;
+                        
+                        zV = getDataLaboratoryValue(n, columnZ);
+                        
+                        if ( zV > maxZvalueFromColumnZ ) {
+                            maxZvalueFromColumnZ = zV; 
+                        }
+                   }
+        }
+        
+        
+        
+        // Compute z-level of each node and store it in the Gephi Data Laboratory. 
+        int maxZLevel = 0;
+        for (int i = 0 ; i< nodeCount; i++ ) {
+                   Node n = nodes[i]; 
+                   double zV = 0f;
+                   if ( columnZ != null ){
+                        zV = getDataLaboratoryValue(n, columnZ);
+                    }
+                   int z_level=0;
+                   if ( maxZvalueFromColumnZ != 0 ) {
+                        z_level = (int) Math.round( (((zV * (double)(zlevels) / maxZvalueFromColumnZ) )));
+                        if ( z_level > maxZLevel ) {
+                            maxZLevel = z_level;
+                        }
+                   }
+                   
+                   n.setAttribute(colResult, z_level);
+                   
+         }
+        
+ 
+        // Detect how many isometric x,y grid scales are needed (z grid scales are on properties user demand)
+        int xGridScales = (int) Math.round(Math.sqrt(nodeCount))+1;
+        int yGridScales = (int) Math.round(Math.sqrt(nodeCount))+1;
+      
+        // Draw isometric network (splitting layers if needed)
+        int k = 0;
+        for (int i = 0; i < xGridScales; i++) {
+         
+            for ( int j=0; j<yGridScales; j++ ) {
+               
+                if ( k < nodeCount ) {
+                  Node node = nodes[k];
+             
+                   int zGrid = 0; // 3D height of each node (z-axis)
+                  
+                   if ( colResult!= null ) {
+                        zGrid= (Integer) (node.getAttribute(colResult));
+                        zGrid*=zdistance;
+                   }
+             
+                   
+                   if ( isHorizontal() ) {
+                        node.setX(yGridToScreen(i,j,(float) ((isReverse()? zGrid: maxZLevel-zGrid))) );
+                        node.setY(xGridToScreen(i,j));
+                   } else {
+                        node.setX(xGridToScreen(i,j));
+                        node.setY(yGridToScreen(i,j,(float) ((isReverse()? zGrid: maxZLevel-zGrid))) );
+                   }
+               
+                   ++k;
+                }
+            }
+            
+        }
+        
+        
+        
+        graph.readUnlock();
+        endAlgo();
+        
+        
+    }
+
+    
+    public double getDataLaboratoryValue(Node n, Column col) {

```

Comment:
Here you could check if the column is a subclass of ```Number``` Then use doubleValue() on the value.
Also, parsing Strings to Double might throw a parse exception. I suggest catching and ignoring that exception since that's not very likely to happen.